### PR TITLE
Receive from closed channel

### PR DESCRIPTION
**Pull request by wangkuiyi**

### CODE REVIEW

**Comment by kavyasrinet** [comment]
Sure, will do.

File: paddle/framework/channel_test.cc [hunk 1]
```diff
@@ -60,6 +60,37 @@ TEST(Channel, SufficientBufferSizeDoesntBlock) {
...
 
+TEST(Channel, ReceiveFromBufferedChannelReturnResidualValuesTest) {
+  const size_t buffer_size = 10;
+  auto ch = MakeChannel<size_t>(buffer_size);
+
+  for (size_t i = 0; i < buffer_size; ++i) {
+    EXPECT_EQ(ch->Send(&i), true);  // sending should not block
+  }
+
+  int out;
```
Comment (chengduoZH) [inline paddle/framework/channel_test.cc]: `int out` ==> `size_t out`

File: paddle/framework/channel_test.cc [hunk 2]
```diff
@@ -119,6 +150,7 @@ TEST(Channel, BufferedChannelCloseUnblocksReceiversTest) {
           int data;
           // All reads should return false
           EXPECT_EQ(ch->Receive(&data), false);
+          EXPECT_EQ(data, 0);
```
Comment (chengduoZH) [inline paddle/framework/channel_test.cc]: I do not think the `data` should be zero when the channel was closed. If the element of channel is `Tensor*` and the channel has been closed, what will the `data` be after `ch->Receive(&data)`

Comment (kavyasrinet) [reply paddle/framework/channel_test.cc]: it should be the zero value of that return type, as mentioned here: https://dave.cheney.net/2014/03/19/channel-axioms
Here are the zero values: https://golang.org/ref/spec#The_zero_value

Comment (wangkuiyi) [reply paddle/framework/channel_test.cc]: The problem here is that C++ syntax doesn't define zero value of all types like Go does. So, basically, we have no idea how to write a C++ code snippet to clear the value of any data type to zero.

File: paddle/framework/channel_test.cc [hunk 3]
```diff
@@ -60,6 +60,38 @@ TEST(Channel, SufficientBufferSizeDoesntBlock) {
...
+
+  for (size_t i = buffer_size / 2; i < buffer_size; ++i) {
+    EXPECT_EQ(ch->Receive(&out),
+              true);  // receving should return residual values.
+    EXPECT_EQ(out, i);
+  }
+
+  for (size_t i = 0; i < buffer_size; ++i) {
+    EXPECT_EQ(ch->Receive(&out),
+              false);  // after receiving residual values, return zeros.
```
Comment (chengduoZH) [inline paddle/framework/channel_test.cc]: This annotation should be removed or changed.
~~`// after receiving residual values, return zeros.`~~

**Pull request merged**